CMakeLists.txt

```diff
@@ -100,6 +100,7 @@ if (MPI_AVAIL)
               -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
               -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
               -DCMAKE_Fortran_COMPILER=${CMAKE_Fortran_COMPILER}
+              -DCMAKE_C_FLAGS="-Wno-error=implicit-function-declaration"
```

Comment:
@Yaraslaut is this a universal flag for all compilers?

Reply:
I believe that they have the same syntax for warnings and errors.
Locally i can not compile with gcc though, getting some errors from highfive

Reply:
Hi @Yaraslaut, thanks for the PR -- I could potentially check this with GCC, which other compilers have you tried it with?

Reply:
Hi @jcoulter12 ,
apparently cmake does not want to reconfigure correctly, I tested `gcc` , and it works
list of compilers that i tested: 
`g++ (GCC) 13.2.1 20230801`  
`clang version 18.0.0` 
`Build cuda_12.2.r12.2/compiler.33191640_0`

I think that you can check  quite new versions of compilers in  github actions with ub-22.04 [see](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md)

Reply:
@anjohan, @jcoulter12 if you feel that it's worth to update GitHub actions I can make a PR for this

Reply:
@anjohan, maybe if @Yaraslaut wants to help, it would be favorable to have a second git actions which is run on PR for a number of standard compiler types? 
Is that possible?